adapters/telaria/telaria.go

```diff
@@ -282,11 +276,15 @@ func (a *TelariaAdapter) MakeBids(internalRequest *openrtb2.BidRequest, external
 	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(bidResp.SeatBid[0].Bid))
 	sb := bidResp.SeatBid[0]
 
-	for _, bid := range sb.Bid {
-		bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
-			Bid:     &bid,
-			BidType: openrtb_ext.BidTypeVideo,
-		})
+	for i := range sb.Bid {
+		bid := sb.Bid[i]
+		if i < len(internalRequest.Imp) {
```

Comment:
If `len(internalRequest.Imp)` is less than `len(sb.Bid)` should we `break` the for loop to optimize a bit?
```
278  
279       for i := range sb.Bid {
280           bid := sb.Bid[i]
281 -         if i < len(internalRequest.Imp) {
    +         if i >= len(internalRequest.Imp) {
    +             break
    +         }
282               bid.ImpID = internalRequest.Imp[i].ID
283               bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
284                   Bid:     &bid,
285                   BidType: openrtb_ext.BidTypeVideo,
286               })
287 -         }
288       }
289       return bidResponse, nil
290   }
291  
adapters/telaria/telaria.go
```

Reply:
Thank you so much for your inputs. They are really good points. I made the suggested changes and added the test cases as well.